bench.py

```diff
@@ -0,0 +1,90 @@
+import time
+from sklearn.decomposition import FastICA
+import numpy as np
+import pandas as pd
+import streamlit as st
+import altair as alt
+import argparse
+
+parser = argparse.ArgumentParser(
+    description="Determine whether to save/load the dataframe."
+)
+parser.add_argument(
+    "--save",
+    type=str,
+    default="",
+    metavar="s",
+    help="Saves the dataframe to the path if provided",
+)
+parser.add_argument(
+    "--load",
+    type=str,
+    default="",
+    metavar="l",
+    help="Loads the dataframe at the path if provided",
+)
+args = parser.parse_args()
+
+
+def main() -> None:
+
+    df = None
+    if args.load:
+        df = on_load(args.load)
+    else:
+        X_shapes = []
+        for i in range(10):
+            X_shapes.extend(
+                (10 * int(10 ** (i / 3)), 10 * int(10 ** (j / 3))) for j in range(10)
+            )
+
+        solvers = ("svd", "eigh")
+        transformers = {
+            s: FastICA(n_components=7, random_state=0, svd_solver=s) for s in solvers
+        }
+        total_reps = len(solvers) * len(X_shapes)
+        count = 0
+        data = []
+        for shape in X_shapes:
+            X = np.random.rand(*shape)
+
+            count += 1
+            start = time.time()
+            transformers["svd"].fit_transform(X)
```

Comment:
I think it would be good to create a new micro benchmark comparing `linalg.eigh(X.T.dot(X))` and `linalg.svd`, so we can test them directly without needing `FastICA`.